connection-loader/src/test/java/com/terracotta/connection/api/URIUtilsTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2011-2017 Software AG, Darmstadt, Germany and/or Software AG USA Inc., Reston, VA, USA, and/or its subsidiaries and/or its affiliates and/or their licensors.
+ * Use, reproduction, transfer, publication or disclosure is prohibited except as specifically provided for in your License Agreement with Software AG.
+ */
+package com.terracotta.connection.api;
+
+import org.junit.Test;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class URIUtilsTest {
```

Comment:
In fact, the is a problem here. You should use a `ExpectedException` rule instead. Because right now we don't know if it's `new URI` or `validateTerracottaURI` which causes the exception. But if you do

```
URI uri = new URI("...");
expectedException.expect(URISyntaxException.class); // and because I'm crazy, I would also expect the message
URIUtils.validateTerracottaURI(uri);
```

It will fail when wanted only

Reply:
I agree except that stylistically I prefer using try catch constructs instead of ExpectedException rules.  To each his own.

Reply:
Ouch... So sad and error prone.

```
URI uri = new URI("...");
try {
  URIUtils.validateTerracottaURI(uri);
  fail("Should throw an exception"); // error prone because forgotten
} catch(URISyntaxException e) {
 assertThat(e.getMessage()).isEqualsTo("...");
}
```
I've fix some errors like that in another OSS framework you know well.

Reply:
Maybe - except that, at a system level, I don't particularly need / want to distinguish between strings rejected by new URI() and validateTerracottaURI().

However I don't mind changing the test to be more explicit, if you will feel happier.

Reply:
Yes, it will make me happier :-)